src/cozmo/action.py

```diff
@@ -219,9 +210,21 @@ def _recv_msg_robot_completed_action(self, evt, *, msg):
         elif result == types.INTERRUPTED:
             self._set_failed('interrupted', 'Action was interrupted')
 
+        # Can add additional if statements for other ActionResults to exposure more specific failure reasons to users
+
         else:
-            self._set_failed('unknown', 'Action failed with unknown reason')
-            logger.error('Received unknown action result status %s', msg)
+            # Got some other result figure out if it is a result that indicates the action can be retried
+            resultCategory = result >> _clad_to_game_cozmo.ARCBitShift.NUM_BITS
+            resultCategories = _clad_to_game_cozmo.ActionResultCategory
+            
+            if resultCategory == resultCategories.ABORT:
+                self._set_failed('aborted', 'Action failed')
```

Comment:
Ideally we would add the result to the message, so e.g. 'Action failed: NO_FACE', but we don't have an automatic enum->string conversion in the python clad generator currently, so you have to build something each time - we can add this later though, so don't worry about it for now (this is mostly just a note for myself).